### DISCUSSION TITLE
Creating table templates with show

### DISCUSSION THREAD

**Discussion by TheJanzap** [PgBiel/typst-tablex#130] [Q&A]
I tried to create a template for all tablex in my document by using `show tablex: tb => {}`, but the compiler returns `Only element functions can be used as selectors`. How can I set a default style for my tables?

**Comment by PgBiel** [comment]
`tablex` is indeed a regular function, much like your usual `calc.rem(5, 3)` function; `tablex` is, however, special in that it returns content you can display in the document. But that's about it. Ideally, `tablex` would be turned into an element, but currently that's impossible in Typst (no syntax for that). So you can't use show rules on tablex at the moment.

One way around it is to simply specify default parameters for `tablex`. For example, you can write:

```typ
// simulate a set rule
#let tablex = tablex.with(stroke: red)
```

Or

```typ
// simulate a show rule
#let old-tablex = tablex
#let tablex(..args) = {
  let it = old-tablex(..args)
  block(stroke: (left: red), it)
}
```

**Reply by TheJanzap** [reply]
The simulated set rule works, but I can't get the show rule to work.
I did find #33, which is the default look I'm trying to achieve, but it looks like I have to set this for every table manually at the moment?
